openedx/features/course_experience/static/course_experience/js/CourseHome.js

```diff
@@ -144,75 +144,13 @@ export class CourseHome {  // eslint-disable-line import/prefer-default-export
     );
   }
 
-  /**
-   * Persists the collapsed state of the upgrade message. If the message is collapsed,
-   * this information is persisted to local storage. Expanding the message *removes* the
-   * key from local storage.
-   */
-  persistUpgradeMessageState(collapsed) {
-    if (window.localStorage) {
-      if (collapsed) {
-        window.localStorage.setItem(this.msgStateStorageKey, true);
-      } else {
-        window.localStorage.removeItem(this.msgStateStorageKey);
-      }
-    }
-  }
-
   configureUpgradeMessage() {
-    const $vcMessage = $('.vc-message');
-    const $vcDismissToggle = $('.vc-toggle', $vcMessage);
     const logEventProperties = { courseRunKey: this.courseRunKey };
 
-    Logger.log('edx.bi.course.upgrade.hero.displayed', logEventProperties);
-
-    // Get height of container and button
-    let vcHeight = $vcMessage.outerHeight();
-
-    // Update based on window
-    window.onresize = () => {
-      if (!$vcMessage.hasClass('polite')) {
-        vcHeight = $vcMessage.outerHeight();
-      }
-    };
-
-    function collapseMessage(duration = 400) {
-      $('.vc-fade').fadeOut(duration, () => {
-        $vcDismissToggle.text(gettext('Show more')).attr('aria-expanded', false);
-        $('.vc-polite-only').fadeIn(duration);
-        $vcMessage.height('auto').addClass('polite');
-      });
-    }
-
-    // Use the previously-persisted state to determine the initial display state of the message.
-    if (window.localStorage && window.localStorage.getItem(this.msgStateStorageKey)) {
-      collapseMessage(0);
-    }
-    $vcMessage.show();
-
-    $vcDismissToggle.click(() => {
-      if ($vcMessage.hasClass('polite')) {
-        // Expand message
-        Logger.log('edx.bi.course.upgrade.hero.expanded', logEventProperties);
-        this.persistUpgradeMessageState(false);
-
-        $('.vc-fade').fadeOut(400);
-        $vcMessage.animate({ height: vcHeight }, 400, () => {
-          $vcMessage.height('auto').removeClass('polite');
-          $vcDismissToggle.text(gettext('Show less')).attr('aria-expanded', true);
-          $('.vc-fade').fadeIn(400);
-        });
-      } else {
-        // Collapse message
-        Logger.log('edx.bi.course.upgrade.hero.collapsed', logEventProperties);
-        this.persistUpgradeMessageState(true);
-        collapseMessage();
-      }
-    });
-
-    $('.btn-upgrade', $vcMessage).click(() => {
-      Logger.log('edx.bi.course.upgrade.hero.clicked', logEventProperties);
-      Logger.log('edx.course.enrollment.upgrade.clicked', { location: 'hero' });
+    Logger.log('edx.bi.course.upgrade.sidebarupsell.displayed', logEventProperties);
+    $('.section-upgrade .btn-upgrade').click(() => {
+      Logger.log('edx.bi.course.upgrade.sidebarupsell.clicked', logEventProperties);
+      Logger.log('edx.course.enrollment.upgrade.clicked', { location: 'sidebar-message' });
```

Comment:
I would include the code to scroll down to the sock and expand it in this PR, even if the sock only shows up when the language is English.

Reply:
what code do you mean? can you clarify?

Reply:
Currently, clicking the 'Learn More' link doesn't do anything, but I was suggesting that we put in the code that would make it do what it's supposed to when the sock work is finalized. I think it just makes more sense to do that in this ticket/PR than the i18n one.

Reply:
I see, i forgot I hadn't added that yet, fixed